AK/CharacterTypes.h

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2021, Max Wipfli <mail@maxwipfli.ch>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Types.h>
+
+// NOTE: For a quick reference for most of this, see https://www.cplusplus.com/reference/cctype/ and https://infra.spec.whatwg.org/#code-points.
+// NOTE: To avoid ambiguity when including this header, all methods contains names should contain "ascii" or "unicode".
+
+namespace AK {
+
+constexpr bool is_ascii(u32 code_point)
+{
+    return code_point < 0x80;
+}
+
+constexpr bool is_ascii_digit(u32 code_point)
+{
+    return code_point >= '0' && code_point <= '9';
+}
+
+constexpr bool is_ascii_upper_alpha(u32 code_point)
+{
+    return (code_point >= 'A' && code_point <= 'Z');
+}
+
+constexpr bool is_ascii_lower_alpha(u32 code_point)
+{
+    return (code_point >= 'a' && code_point <= 'z');
+}
+
+constexpr bool is_ascii_alpha(u32 code_point)
+{
+    return is_ascii_lower_alpha(code_point) || is_ascii_upper_alpha(code_point);
+}
+
+constexpr bool is_ascii_alphanumeric(u32 code_point)
+{
+    return is_ascii_alpha(code_point) || is_ascii_digit(code_point);
+}
+
+constexpr bool is_ascii_hex_digit(u32 code_point)
+{
+    return is_ascii_digit(code_point) || (code_point >= 'A' && code_point <= 'F') || (code_point >= 'a' && code_point <= 'f');
+}
+
+constexpr bool is_ascii_blank(u32 code_point)
+{
+    return code_point == '\t' || code_point == ' ';
+}
+
+constexpr bool is_ascii_space(u32 code_point)
+{
+    return code_point == ' ' || code_point == '\t' || code_point == '\n' || code_point == '\v' || code_point == '\f' || code_point == '\r';
```

Comment:
would it make sense to have this simplify to

`is_ascii_blank(code_point) || code_point == '\n' || code_point == '\v' || code_point == '\f' || code_point == '\r'` 
?

Reply:
While I am very much in favor of code reuse, I think this just makes it more convoluted than it needs to be, especially because `is_ascii_blank` is so trivial. If it would check for more than just space and tab, sure. I know this is more a question of style, but if you quickly want to know which characters the function considers to be whitespace, it is much more obvious the way I wrote it.

Reply:
fair.